### PR TITLE
feat(system_launch): replace system launch

### PR DESCRIPTION
**Pull request by asana17**
Signed-off-by: asana17 [akihiro.sakurai@tier4.jp](mailto:akihiro.sakurai@tier4.jp)

## PR Type

- Improvement

## Related Links

https://github.com/autowarefoundation/autoware.universe/pull/2840
https://github.com/tier4/aip_launcher/pull/101
[TIER IV INTERNAL LINK TO SLACK](https://star4.slack.com/archives/C4P0NSMB5/p1675045362108509?thread_ts=1672881973.897619&cid=C4P0NSMB5)

## Description
Following changes are made by this PR. 
1. add launch configs of  dummy_diag_publisher
2. move system_launch to autoware_launch (use tier4_system_launch instead of system_launch)
 
I tested launch files by using the pilot-auto repository (https://github.com/tier4/pilot-auto).  
Please see the links above for more details.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: autoware_launch/package.xml [hunk 1]
```diff
@@ -6,16 +6,21 @@
...
   <maintainer email="isamu.takagi@tier4.jp">Takagi, Isamu</maintainer>
+  <maintainer email="akihiro.sakurai@tier4.jp">Akihiro Sakurai</maintainer>
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
 
   <exec_depend>ad_api_adaptors</exec_depend>
+  <exec_depend>component_state_monitor</exec_depend>
+  <exec_depend>dummy_diag_publisher</exec_depend>
+  <exec_depend>emergency_handler</exec_depend>
```
Comment (kminoda) [inline autoware_launch/package.xml]: Do we need these dependency to individual packages? I think it is sufficient to just write `tier4_system_launch`.

File: system_launch/launch/system.launch.xml [hunk 2]
```diff
@@ -1,84 +0,0 @@
...
-    </group>
-
-    <!-- Dummy Diag Publisher -->
-    <group if="$(eval &quot;'$(var run_mode)'=='planning_simulation'&quot;)">
-      <push-ros-namespace namespace="dummy_diag_publisher"/>
-      <include file="$(find-pkg-share system_launch)/launch/dummy_diag_publisher/control.launch.xml"/>
-      <include file="$(find-pkg-share system_launch)/launch/dummy_diag_publisher/localization.launch.xml"/>
-      <include file="$(find-pkg-share system_launch)/launch/dummy_diag_publisher/system.launch.xml"/>
-      <include file="$(find-pkg-share system_launch)/launch/dummy_diag_publisher/vehicle.launch.xml"/>
-      <include file="$(var sensor_launch_pkg)/launch/dummy_diag_publisher/sensing.launch.xml"/>
```
Comment (kminoda) [inline system_launch/launch/system.launch.xml:80]: Where is this set of dummy diags for sensors will be launched in the modified version? (Apparently it is missing?)

File: autoware_launch/launch/components/tier4_system_component.launch.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    <arg name="emergency_handler_param_path" value="$(find-pkg-share autoware_launch)/config/system/emergency_handler/emergency_handler.param.yaml"/>
+    <arg name="mrm_comfortable_stop_operator_param_path" value="$(find-pkg-share autoware_launch)/config/system/mrm_comfortable_stop_operator/mrm_comfortable_stop_operator.param.yaml"/>
+    <arg name="mrm_emergency_stop_operator_param_path" value="$(find-pkg-share autoware_launch)/config/system/mrm_emergency_stop_operator/mrm_emergency_stop_operator.param.yaml"/>
+    <arg name="system_error_monitor_param_path" value="$(find-pkg-share autoware_launch)/config/system/system_error_monitor/system_error_monitor.param.yaml"/>
+    <arg name="system_error_monitor_planning_simulator_param_path" value="$(find-pkg-share autoware_launch)/config/system/system_error_monitor/system_error_monitor.planning_simulation.param.yaml"/>
+    <arg name="diagnostic_aggregator_system_param_path" value="$(find-pkg-share autoware_launch)/config/system/diagnostic_aggregator/system.param.yaml"/>
+    <arg name="diagnostic_aggregator_vehicle_param_path" value="$(find-pkg-share autoware_launch)/config/system/diagnostic_aggregator/vehicle.param.yaml"/>
```
Comment (kminoda) [inline autoware_launch/launch/components/tier4_system_component.launch.xml:22]: Minor question: Do you know why this diagnostic_aggregator parameter files are stored in a different structure from awf/autoware_launch?
- awf/autoware_launch: `config/system/system_error_monitor/diagnostic_aggregator/vehicle.param.yaml`
- tier4/autoware_launch: `config/system/diagnostic_aggregator/vehicle.param.yaml`

Considering the sync from awf/autoware_launch -> tier4/autoware_launch which will be introduced after this PR, we want to mitigate unnecessary diffs as much as possible (not necessarily in this PR, though)

**Review by kminoda — COMMENTED** [review]
Thank you very much! This PR is exactly what we wanted.
I left some small comments.

File: system_launch/launch/system.launch.xml (hunk 2 above)
Comment (asana17) [reply system_launch/launch/system.launch.xml:80]: I moved the top three parameters to one parameter file, [autoware_launch/config/system/dummy_diag_publisher/dummy_diag_publisher.param.yaml](https://github.com/tier4/autoware_launch/pull/762/files#diff-80fcd5260bc3f3b8ba9281f0ac3121342d40583115474c15557d85ecc3af1731).


The last params (for sensors) are originally in [tier4/aip_launcher](https://github.com/tier4/aip_launcher/blob/tier4/universe/aip_xx1_launch/launch/dummy_diag_publisher/sensing.launch.xml). I moved these params into [tier4/aip_laucher:<sensor_name>/config/dummy_diag_publisher/sensor_kit.param.yaml](https://github.com/tier4/aip_launcher/pull/101/files#diff-90a7719958a3cd564dca01a51f93327c1ef0fb73f4facc5bacafb09be1093bb5).

File: autoware_launch/package.xml (hunk 1 above)
Comment (asana17) [reply autoware_launch/package.xml]: You're right. I removed those unnecessary dependencies. [autoware_launch/package.xml](https://github.com/tier4/autoware_launch/pull/762/files#diff-1bd8e4504c256fbd05229b25dedce131f727aa9f2d291aeb504f987d8d6bf157)

File: autoware_launch/launch/components/tier4_system_component.launch.xml (hunk 3 above)
Comment (asana17) [reply autoware_launch/launch/components/tier4_system_component.launch.xml:22]: @kminoda It is not clear why this structural difference exists, but this is not critical for the system_error_monitor behavior. I also asked @ito-san to confirm this structural difference is not essential.

File: autoware_launch/launch/autoware.launch.xml [hunk 4]
```diff
@@ -65,8 +66,9 @@
 
   <!-- System -->
   <group if="$(var launch_system)">
-    <include file="$(find-pkg-share system_launch)/launch/system.launch.xml">
+    <include file="$(find-pkg-share autoware_launch)/launch/components/tier4_system_component.launch.xml">
       <arg name="run_mode" value="$(var system_run_mode)"/>
+      <arg name="launch_system_monitor" value="$(var launch_system_monitor)"/>
       <arg name="sensor_model" value="$(var sensor_model)"/>
```
Comment (takayuki5168) [inline autoware_launch/launch/autoware.launch.xml]: nits: better to remove all the arguments here like this change https://github.com/autowarefoundation/autoware_launch/pull/185/files.

This modification was adopted based on the TIER IR's internal discussion.
https://star4.slack.com/archives/C4P0NSMB5/p1674616811421119

Comment (takayuki5168) [reply autoware_launch/launch/autoware.launch.xml]: For me, it's okay to deal with this comment with another PR once this PR is merged.

File: system_launch/launch/system.launch.xml (hunk 2 above)
Comment (kminoda) [reply system_launch/launch/system.launch.xml:80]: Sorry, my question didn't make sense.
What I wanted to ask: Is this set of dummy diags for sensors also launched in the modified version? (Apparently it is missing?)

And seems I found the answer by myself, it was launched in teir4_system_launch. So never mind!

File: autoware_launch/launch/components/tier4_system_component.launch.xml (hunk 3 above)
Comment (kminoda) [reply autoware_launch/launch/components/tier4_system_component.launch.xml:22]: OK, thanks!

File: autoware_launch/launch/autoware.launch.xml (hunk 4 above)
Comment (asana17) [reply autoware_launch/launch/autoware.launch.xml]: OK, I will handle it in another PR.

**Review by kminoda — APPROVED** [review]
LGTM
(Although the comment https://github.com/tier4/autoware_launch/pull/762#discussion_r1099478185 is not yet addressed, I think we can merge these PRs first and then discuss these differences)

**Pull request merged**